verilog/formatting/tree_unwrapper.cc

```diff
@@ -1294,6 +1336,19 @@ void TreeUnwrapper::ReshapeTokenPartitions(
     case NodeEnum::kStatement: {
       // This handles cases like macro-calls followed by a semicolon.
       AttachTrailingSemicolonToPreviousPartition(&partition);
+
+      const auto* subtree = GetSubtreeAsSymbol(node, NodeEnum::kStatement, 0);
+      if (subtree != nullptr) {
+        const auto& subnode = SymbolCastToNode(*subtree);
+        if (NodeEnum(subnode.Tag().tag) == NodeEnum::kReferenceCallBase) {
+          if (partition.Children().size() >= 2) {
+            const auto& uwline = partition.Children()[1].Value();
+            if (uwline.PartitionPolicy() == PartitionPolicyEnum::kAppendFittingSubPartitions) {
+              verible::MergeLeafIntoNextLeaf(&partition.Children()[0]);
+            }
+          }
```

Comment:
In general, let's define these CST node accessors as helper functions in the CST (which can then be unit-tested).  It is hard to infer the intent in this form.